File: frontend/src/types/stats.ts
export type RecipeSourceStat = { source_domain: string; count: number };
export type TopRecipeStat = { recipe_id: number; title: string; appearance_count: number };
export type TopIngredientStat = { name: string; count: number };

export type StatsOut = {
  total_recipes: number;
  total_lists: number;
  recipes_by_source: RecipeSourceStat[];
  top_recipes_in_lists: TopRecipeStat[];
  top_ingredients: TopIngredientStat[];
};

File: backend/app/schemas/stats.py
from __future__ import annotations

from pydantic import BaseModel


class RecipeSourceStat(BaseModel):
    source_domain: str
    count: int


class TopRecipeStat(BaseModel):
    recipe_id: int
    title: str
    appearance_count: int


class TopIngredientStat(BaseModel):
    name: str
    count: int


class StatsOut(BaseModel):
    total_recipes: int
    total_lists: int
    recipes_by_source: list[RecipeSourceStat]
    top_recipes_in_lists: list[TopRecipeStat]
    top_ingredients: list[TopIngredientStat]
